### PR TITLE
change reload mods logic to have weapon reparsing work better on conn…

### PR DESCRIPTION
**Pull request by BobTheBob9**
…ect to a server with different enabled mods

### CODE REVIEW

File: Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut [hunk 1]
```diff
@@ -7,16 +7,15 @@ global function ThreadedAuthAndConnectToServer
 
 // Stop peeking
 
-const int BUTTONS_PER_PAGE = 15 // Number of servers we show
-const float DOUBLE_CLICK_TIME_MS = 0.4 // Max time between clicks for double click registering 
+const int BUTTONS_PER_PAGE = 15
+const float DOUBLE_CLICK_TIME_MS = 0.4 // unsure what the ideal value is
```
Comment (EladNLG) [inline Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut:11]: 0.5s is the time used for windows by default.

File: Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut [hunk 2]
```diff
@@ -91,10 +86,9 @@ struct {
 
 
 
-bool function FloatsEqual( float arg1, float arg2, float epsilon )
+bool function floatCompareInRange(float arg1, float arg2, float tolerance)
 {
-	if ( fabs( arg1 - arg2 ) < epsilon ) return true
-	
+	if ( arg1 > arg2 - tolerance && arg1 < arg2 + tolerance) return true
```
Comment (EladNLG) [inline Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut:91]: The old version of this function worked just fine? No need to change it...

File: Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut [hunk 3]
```diff
@@ -285,7 +293,7 @@ void function SliderBarUpdate()
 	Hud_SetPos( sliderPanel , 2, newPos )
 	Hud_SetPos( movementCapture , 2, newPos )
 
-	file.scrollOffset = -int( ( ( newPos - minYPos ) / useableSpace ) * ( file.serversArrayFiltered.len() - BUTTONS_PER_PAGE ) )
+	file.scrollOffset = -int( ( (newPos - minYPos) / useableSpace ) * (file.serversArrayFiltered.len() - 15) )
```
Comment (EladNLG) [inline Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut:296]: Use BUTTONS_PER_PAGE.

File: Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut [hunk 4]
```diff
@@ -295,10 +303,10 @@ void function UpdateListSliderHeight( float servers )
 	var sliderPanel = Hud_GetChild( file.menu , "BtnServerListSliderPanel" )
 	var movementCapture = Hud_GetChild( file.menu , "MouseMovementCapture" )
 
-	float maxHeight = 562.0 * ( GetScreenSize()[1] / 1080.0 )
-	float minHeight = 80.0 * ( GetScreenSize()[1] / 1080.0 )
+	float maxHeight = 562.0 * (GetScreenSize()[1] / 1080.0)
+	float minHeight = 80.0 * (GetScreenSize()[1] / 1080.0)
 
-	float height = maxHeight * ( BUTTONS_PER_PAGE / servers )
+	float height = maxHeight * (15.0 / servers )
```
Comment (EladNLG) [inline Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut:309]: Same issue here.

File: Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut [hunk 5]
```diff
@@ -329,7 +339,7 @@ void function UpdateListSliderPosition( int servers )
 
 void function OnScrollDown( var button )
 {
-	if (file.serversArrayFiltered.len() <= BUTTONS_PER_PAGE) return
+	if (file.serversArrayFiltered.len() <= 15) return
```
Comment (EladNLG) [inline Northstar.Client/mod/scripts/vscripts/ui/menu_ns_serverbrowser.nut:342]: And here.

**Review by EladNLG — CHANGES_REQUESTED** [review]
Many unnecessary changes

**Comment by BobTheBob9** [comment]
this sucks atm and should be done in native code, not script

**Pull request closed without merging**